### PR TITLE
Story 2

### PR DESCRIPTION
**Pull request by JamesMcGraaw**

### CODE REVIEW

File: index-test.php
Comment (bmiller180787) [inline index-test.php:1]: You've got quite a few lines of Comments - are they needed?

File: add-deck.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+<header>
+    <a href="index.php">Go back home</a>
+</header>
+
+<h1>
+    Add a new deck:
+</h1>
+
+<div class="add-deck">
+    <form action="index.php" method="post">
```
Comment (ashleycoles) [inline add-deck.php:47]: Suggest adding some validation attributes to this form - for example you can use `required` to make fields non optional. There are a few examples here: https://www.w3schools.com/html/html_form_attributes.asp

File: add-deck.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        </p>
+        <p>
+            <label>Enter a URL for a deck image (HTTPS):</label>
+            <input name="image" id="image" placeholder="https://www.google.com/imghp?hl=en"
+                   pattern="https://.*"><br>
+        </p>
+        <p>
+            <label>Enter a URL for Moxfield (HTTPS):</label>
+            <input name="moxfield_link" id="moxfield-link" placeholder="https://moxfield.com"
+                   pattern="https://.*"><br>
```
Comment (ashleycoles) [inline add-deck.php]: br tags should only ever be used to create new lines in a block of text - better to put the spacing in using CSS instead.

File: index.php [hunk 3]
```diff
@@ -1,8 +1,13 @@
 <?php
 require_once 'DeckDAO.php';
 require_once 'functions.php';
-$deckDao = new DeckDao();
-$decks = $deckDao->fetchAll();
+$deckDAO = new DeckDAO();
+if (isset($_POST['name_of_deck'])) {
+    $newDeck = new Deck($_POST['name_of_deck'], $_POST['format'], $_POST['colourid'], $_POST['archetype']
```
Comment (ashleycoles) [inline index.php]: Worth adding some basic checks to make sure you actually have all the data - IE they haven't left fields empty

**Review by ashleycoles — CHANGES_REQUESTED** [review]
Really great work here James, especially your tests! Only comments are relating to adding some form validation.

File: add-deck.php (hunk 2 above)
Comment (JamesMcGraaw) [reply add-deck.php]: Had actually removed all my <br> tags earlier but missed those ones!

File: add-deck.php (hunk 1 above)
Comment (JamesMcGraaw) [reply add-deck.php:47]: Added required fields for those that can't have a value of null in DB

File: index.php (hunk 3 above)
Comment (JamesMcGraaw) [reply index.php]: Added something to check if it contains strings with length > 0.

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**